### DISCUSSION TITLE
Benvenuti nelle discussioni della traduzione di Stellaris in italiano

### DISCUSSION THREAD

**Discussion by Neurone** [Neurone/stellaris-italian-translation#18] [General]
Benvenuti! Per qualunque dubbio o domande su come contribuire alla mod non esitate a chiedere.

**Comment by Nefando** [comment]
Ciao Neurone, 
ho ricominciato di recente a giocare a Stellaris insieme ad alcuni amici ed abbiamo inserito nel playset la tua mod della traduzione in italiano: ottimo lavoro!, ha permesso di allargare la cerchia ai giocatori meno fluenti con la lingua inglese.

Giocando ho notato qualche piccolo refuso e qualche errore di sintassi che restituisce solo la chiave (in particolare negli eventi). Ho già visto dove intervenire e appena posso applico le correzioni, inviando successivamente una pull request. E' passato un po' dall'ultima volta che ho bazzicato GitHub, appena tolta la ruggine mi metto all'opera.

Prima però, e anche in previsione (tempo permettendo) di poter contribuire con qualche traduzione, volevo chiederti quali file non vanno modificati per poter mantenere attivo l'iron mode e gli achievement. Ho letto su Steam che la traduzione non è completa proprio per ottenere questo scopo, per cui prima di mettere mano in modo dannoso preferisco chiedere. Puoi anche indicarmi in alternativa se ci sono file corposi che possono essere tradotti integralmente senza rischi. 

Di nuovo grazie per il buon lavoro, attendo tue indicazioni.

**Reply by Neurone** [reply]
Benvenuto Nefando, mi fa piacere che il mod sia utile e che tu voglia contribuire a correggere e aggiungere traduzioni. Ti ringrazio in anticipo :)

Tutto quello che trovi qui su GitHub è già traducibile senza rischi di disattivare gli achievement e l'iron mode, e in particolare i file che verranno sovrapposti a quelli del gioco si trovano tutti all'interno del percorso **/src/mod**.

Se dovesse servire invece aggiungere un file non ancora presente sul repository, prima di modificarlo bisogna assicurarsi che non sia incluso nella lista di file che vengono controllati all'avvio del gioco. I file che indica questa lista si chiama `checksum_manifest.txt` e lo trovi all'interno della cartella di gioco. Attualmente il contenuto del file è il seguente:

```
directory 
name = common
sub_directories = yes
file_extension = .txt

directory 
name = common
sub_directories = yes
file_extension = .lua

directory
name = common
sub_directories = yes
file_extension = .csv

directory
name = events
sub_directories = yes
file_extension = .txt

directory
name = map
sub_directories = yes
file_extension = .lua

directory
name = map
sub_directories = yes
file_extension = .txt

directory
name = localisation_synced
sub_directories = yes
file_extension = .yml
```

Prendendo ad esempio la prima sezione, significa che verranno controllati tutti i file `.txt` all'interno della cartella `common` e tutte le sue sotto cartelle. Se anche solo un file di questo tipo viene modificato, il controllo fallisce e il gioco impedisce l'attivazione di iron mode e achievement. E lo stesso vale per tutte le altre sezioni indicate nel file di checksum.

**Reply by Nefando** [reply]
Grazie mille per la spiegazione; mi sono accorto solo successivamente che lo avevi già scritto in prima pagina, mea culpa. 

Ho notato una cosa che però non so come risolvere tramite GitHub: il contenuto del file _diplo_stances_l_english.yml_ non viene visualizzato (restituisce le chiavi), per cui ho fatto una prova e credo che il problema sia che non è codificato UTF-8-BOM. Infatti aprendolo e salvandolo dopo aver applicato il BOM sembra che tutto funzioni regolarmente. E' possibile risolvere questo problema tramite una pull request oppure deve essere caricato ex novo nel repository? (abbi pazienza, confesso che me ne sto approfittando per imparare qualcosa :) )

**Reply by Neurone** [reply]
In realtà ho aggiunto la descrizione nella pagina principale dopo che hai fatto la domanda, quindi grazie a te perché così miglioriamo le istruzioni per tutti :)

Per la questione della modifica del BOM, non è possibile farlo da interfaccia web di GitHub. Serve farlo in locale con un editor di testo un po' evoluto (Notepad++ o Visual Studio Code vanno benissimo) e poi serve un po' di manualità con git per gestire la PR. Sembra complicato a dirsi, ma una volta presa la mano è molto semplice a farsi.

In sintesi qui i passi da seguire:
1. fork da sito github di questo repo nel tuo repo github
2. clone del tuo repo github
3. modifica del file in locale
4. commit in locale
5. push sul tuo repo remoto
6. da sito github sul tuo repo apparirà la possibilità di proporre la pull request

Io ti consiglio di usare Visual Studio Code, che oltre con la modifica dei file ti aiuta anche con l'autenticazione su GitHub senza che fai tutto da riga di comando, se non sei pratico.

Ti lascio un po' di tempo per provare, ma se non riesci o non hai tempo faccio io la modifica.
